AbletonMCP_Remote_Script/handlers/devices.py

```
@@ -0,0 +1,1380 @@
+            ctrl.log_message("Error setting macro value: " + str(e))
+        raise
+
+
+# --- Drum Pad Operations ---
+
+
+def _get_drum_rack(song, track_index, device_index):
+    """Resolve a drum rack device, raising if not found or not a drum rack."""
+    track = get_track(song, track_index)
+    if device_index < 0 or device_index >= len(track.devices):
+        raise IndexError("Device index out of range")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)